django/website/logframe/views.py

```diff
@@ -58,3 +62,31 @@ def get_data(self, logframe, data):
             'result': ResultSerializer(self.object).data,
         })
         return data
+
+
+class CreateLogframe(PermissionRequiredMixin, CreateView):
+    model = LogFrame
+    fields = ['name']
+    template_name = 'logframe/create_logframe.html'
+    permission_required = 'logframe.edit_logframe'
+
+    def get_unique_slug_name(self, logframe):
+        slug = logframe.name.lower()
+        slug = slug.replace(' ', '_')
+        slug = re.sub('[^\w-]', '', slug)
+        slug = slug[:46]
+        if LogFrame.objects.filter(slug=slug).exists():
+            count = LogFrame.objects.filter(slug__startswith=slug).count() + 1
+            slug += unicode(count)
```

Comment:
I'm wondering why this isn't just generated automatically on the model rather than in the view. Would this avoid the need for duplicate code in the migration?
This will also fail if you try to create log frames with names "my_test3", "my test" and "my_test". Why not generate the slug first, test another log frame does not have the same slug and increment the suffix until it is unique.